store/mongo/migration_2_0_0.go

```diff
@@ -104,50 +93,59 @@ func (m *migration_2_0_0) Up(from migrate.Version) error {
 	ctx = identity.WithContext(ctx, &identity.Identity{
 		Tenant: tenantID,
 	})
+	writes := make([]mongo.WriteModel, 0, findBatchSize)
 
-	for collection := range collections {
-		// get all the documents in the collection
+	for collection, idxes := range collections {
+		writes = writes[:0]
 		findOptions := mopts.Find().
 			SetBatchSize(findBatchSize).
 			SetSort(bson.D{{Key: dbFieldID, Value: 1}})
 		coll := client.Database(m.db).Collection(collection)
 		collOut := client.Database(DbName).Collection(collection)
-		_, err := collOut.Indexes().CreateMany(ctx, collections[collection].Indexes)
-		if err != nil {
-			return err
+		if len(idxes.Indexes) > 0 {
+			_, err := collOut.Indexes().CreateMany(ctx, collections[collection].Indexes)
+			if err != nil {
+				return err
+			}
+		}
+		if m.db == DbName {
+			_, err := coll.UpdateMany(ctx, bson.D{
+				{Key: mstore.FieldTenantID, Value: bson.D{
+					{Key: "$exists", Value: false},
+				}},
+			}, bson.D{{Key: "$set", Value: bson.D{
+				{Key: mstore.FieldTenantID, Value: ""},
+			}}},
+			)
+			if err != nil {
+				return err
+			}
+			continue
 		}
 
+		// get all the documents in the collection
 		cur, err := coll.Find(ctx, bson.D{}, findOptions)
 		if err != nil {
 			return err
 		}
 		defer cur.Close(ctx)
 
-		writes := make([]mongo.WriteModel, 0, findBatchSize)
-
 		// migrate the documents
 		for cur.Next(ctx) {
-			item := bson.D{}
+			var item bson.M
```

Comment:
I changed it to a map so I wouldn't have to recreate it later when I retrieved the `_id` key below. In general I would prefer `bson.D` because of the lower complexity of the data type.

Reply:
Reverted to `bson.D` and used the raw bson document to lookup the `_id` field.